tests/task/test_predicates.py

```diff
@@ -0,0 +1,674 @@
+import unittest
+from typing import List, Tuple
+import random
+
+from tests.testhelpers import ScriptedAgentTestConfig, provide_item
+from tests.testhelpers import change_spawn_pos as change_agent_pos
+
+from scripted.baselines import Sleeper
+
+from nmmo.entity.entity import EntityState
+from nmmo.systems import item as Item
+from nmmo.systems import skill as Skill
+from nmmo.lib import material as Material
+from nmmo.lib.log import EventCode
+
+# pylint: disable=import-error
+from nmmo.core.env import Env as TaskEnv
+from nmmo.task.task_api import Repeat, MultiTask
+from nmmo.task.predicate import Predicate
+from nmmo.task.group import Group
+import nmmo.task.predicate.base_predicate as bp
+import nmmo.task.predicate.item_predicate as ip
+import nmmo.task.predicate.gold_predicate as gp
+
+# use the constant reward of 1 for testing predicates
+REWARD = 1
+NUM_AGENT = 6
+ALL_AGENT = Group(list(range(1, NUM_AGENT+1)), 'All')
+
+class TestBasePredicate(unittest.TestCase):
+  # pylint: disable=protected-access,invalid-name
+
+  def _get_taskenv(self,
+                   test_tasks: List[Tuple[Predicate, int]],
+                   grass_map=False):
+
+    config = ScriptedAgentTestConfig()
+    config.PLAYERS = [Sleeper]
+    config.PLAYER_N = NUM_AGENT
+    config.IMMORTAL = True
+
+    tasks = MultiTask(
```

Comment:
Keep seeing MultiTask ... more like a part of env._compute_rewards() then a separate task class